iml-manager-cli/src/main.rs

```diff
@@ -172,20 +191,60 @@ fn main() {
                             purge_duration: pd
                         }),
                     )
+                    .and_then(|resp| iml_manager_client::concat_body(resp))
+                    .map(|(resp, chunk)| {
+                        let status = resp.status();
+                        match status.as_u16() {
+                            200...299 => Ok(serde_json::from_slice::<CmdWrapper>(&chunk)
+                                .expect("Couldn't parse CmdWrapper.")),
+                            400...499 => {
+                                let validation_message: ValidationMessage =
+                                    serde_json::from_slice(&chunk).expect("Could not parse chunk.");
+
+                                Err(validation_message)
+                            }
+                            _ => Err(ValidationMessage {
+                                code: RunStratagemCommandResult::ServerError,
+                                message: "Internal server error.".to_string(),
+                            }),
+                        }
+                    })
                 };
 
-                let CmdWrapper { command }: CmdWrapper = run_cmd(fut).expect("Could not run cmd");
+                let command: Result<CmdWrapper, ValidationMessage> =
+                    run_cmd(fut).expect("Could not run command.");
 
-                let stop_spinner = start_spinner(&command.message);
+                match command {
+                    Ok(CmdWrapper { command }) => {
+                        let stop_spinner = start_spinner(&command.message);
 
-                let command =
-                    run_cmd(api_utils::wait_for_cmd(command)).expect("Could not poll command");
+                        let command = run_cmd(api_utils::wait_for_cmd(command))
+                            .expect("Could not poll command");
 
-                stop_spinner(None);
+                        stop_spinner(None);
 
-                display_cmd_state(&command);
+                        display_cmd_state(&command);
 
-                exit(exitcode::OK)
+                        exit(exitcode::OK)
+                    }
+                    Err(validation_message) => {
+                        let command = Command {
```

Comment:
Instead of constructing a virtual command just to display the message, you should refactor `display_cmd_state` so that it calls different message fns for success / failure etc...